### PR TITLE
Rename InterfaceManagers to standard convention

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Addresses one part of #2178 -- specifically the part regarding InterfaceManagers

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/Systems/Cce/Actions/BoundaryComputeAndSendToEvolution.hpp [hunk 1]
```diff
@@ -112,10 +112,10 @@ struct BoundaryComputeAndSendToEvolution<H5WorldtubeBoundary<Metavariables>,
 /*!
  * \ingroup ActionsGroup
  * \brief Submits a request for CCE boundary data at the specified `time` to the
- * `Cce::GhWorldtubeInterfaceManager`, and sends the data to the
+ * `Cce::GhInterfaceManager`, and sends the data to the
```
Comment (nilsvu) [inline src/Evolution/Systems/Cce/Actions/BoundaryComputeAndSendToEvolution.hpp]: Cce::InterfaceManagers::GhInterfaceManager

(also change below)

File: src/Evolution/Systems/Cce/InterfaceManagers/GhLockstep.cpp [hunk 2]
```diff
@@ -12,26 +12,26 @@
...
 }
 
-void GhLockstepInterfaceManager::insert_gh_data(
+void GhLockstep::insert_gh_data(
     TimeStepId time_id, tnsr::aa<DataVector, 3> spacetime_metric,
     tnsr::iaa<DataVector, 3> phi, tnsr::aa<DataVector, 3> pi,
     const tnsr::aa<DataVector, 3> /*dt_spacetime_metric*/,
     const tnsr::iaa<DataVector, 3> /*dt_phi*/,
     const tnsr::aa<DataVector, 3> /*dt_pi*/) noexcept {
+  // NOLINTNEXTLINE(performance-move-const-arg)
```
Comment (nilsvu) [inline src/Evolution/Systems/Cce/InterfaceManagers/GhLockstep.cpp:27]: Probably better to move down one line

**Review by nilsvu — CHANGES_REQUESTED** [review]

File: src/Evolution/Systems/Cce/InterfaceManagers/GhLockstep.cpp (hunk 2 above)
Comment (moxcodes) [reply src/Evolution/Systems/Cce/InterfaceManagers/GhLockstep.cpp:27]: Sorry, I don't understand -- the clang-tidy error is about the `time_id` that it characterizes as trivially copyable, which is in line 28.

Comment (nilsvu) [reply src/Evolution/Systems/Cce/InterfaceManagers/GhLockstep.cpp:27]: This just displayed strangely on mobile, showed a blank line, just ignore the comment :)

**Review by kidder — COMMENTED** [review]
@nilsdeppe and I have no further requested changes

**Comment by moxcodes** [comment]
fixup posted.
Cheers!

File: src/Evolution/Systems/Cce/Actions/ReceiveGhWorldtubeData.hpp [hunk 3]
```diff
@@ -29,11 +29,11 @@ namespace Actions {
 /*!
  * \ingroup ActionsGroup
  * \brief Stores the boundary data from the GH evolution in the
- * `Cce::GhWorldtubeInterfaceManager`, and sends to the `EvolutionComponent`
+ * `Cce::GhInterfaceManager`, and sends to the `EvolutionComponent`
```
Comment (nilsvu) [inline src/Evolution/Systems/Cce/Actions/ReceiveGhWorldtubeData.hpp]: Here are a few more occurrences. I'm ok with squashing these docs fixes directly

**Comment by moxcodes** [comment]
fixups squashed in, along with the last couple of dox fixes.
Cheers!

**Review by nilsdeppe — APPROVED** [review]

**Review by nilsvu — APPROVED** [review]

**Comment by nilsdeppe** [comment]
clang-tidy failures are being addressed in other PRs.

**Pull request merged**